src/lib/utils/letterDataUtils.ts

```diff
@@ -47,3 +47,47 @@ export const makeWordUsed = (letterArray: LetterData[], validWord: string) => {
 
 	return letterArray;
 };
+
+export const provideHint = (
+	playedLetters: LetterData[],
+	unplayedLetters: LetterData[],
+	wordToSolve: string
+) => {
+	const currentWordFirstIndex = playedLetters.findIndex((letter) => letter.state == 'firstLetter');
+	let hintLetterIndex = playedLetters.findIndex((letter) => letter.state == 'inPlay');
+
+	if (hintLetterIndex == -1) {
+		hintLetterIndex = playedLetters.length;
+	}
+
+	[playedLetters, unplayedLetters] = resetBoard(playedLetters, unplayedLetters);
+
+	const hintLetter = wordToSolve[hintLetterIndex - currentWordFirstIndex];
```

Comment:
`currentWordFirstIndex` can't ever be undefined since `findIndex()` will always return a number (-1 if failure). With that said im not accounting for if it returns -1 so ill make a check for that.

Reply:
donezo